server/sonar-web/src/main/js/apps/maintenance/components/App.tsx

```diff
@@ -0,0 +1,285 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { getMigrationStatus, getSystemStatus, migrateDatabase } from '../../../api/system';
+import DateFromNow from '../../../components/intl/DateFromNow';
+import TimeFormatter from '../../../components/intl/TimeFormatter';
+import { translate } from '../../../helpers/l10n';
+import { getBaseUrl } from '../../../helpers/urls';
+import '../styles.css';
+
+interface Props {
+  // eslint-disable-next-line camelcase
+  location: { query: { return_to?: string } };
+  setup: boolean;
+}
+
+interface State {
+  message?: string;
+  startedAt?: string;
+  state?: string;
+  status?: string;
+  wasStarting?: boolean;
+}
+
+export default class App extends React.PureComponent<Props, State> {
+  interval?: number;
+  mounted: boolean = false;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Type boolean trivially inferred from a boolean literal, remove type annotation ([typescript:S3257](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3257))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fmaintenance&issues=AWGO4PeQFMFpNJFuft7N&open=AWGO4PeQFMFpNJFuft7N)

[ISSUE_KEY]: AWGO4PeQFMFpNJFuft7N